src/main/scala/com/campudus/tableaux/database/DatabaseQuery.scala

```diff
@@ -147,150 +153,99 @@ class ColumnStructure(transaction: DatabaseConnection) {
                     |    ON DELETE CASCADE
                     |)""".stripMargin, Json.arr())
-    id <- Future.successful { result.getArray("results").get[JsonArray](0).get[Long](0) }
     _ <- t.commit()
-  } yield id
+  } yield result.get[JsonArray](0).get[Long](0)
 
-  def get(table: Table, columnId: IdType): Future[JsonArray] = for {
-    t <- transaction.begin()
-    (t, result) <- t.query("""
+  def get(tableId: IdType, columnId: IdType): Future[JsonArray] = for {
+    result <- connection.singleQuery("""
                               |SELECT column_id, user_column_name, column_type
                               |  FROM system_columns
                               |  WHERE table_id = ? AND column_id = ?
-                              |  ORDER BY column_id""".stripMargin, Json.arr(table.id, columnId))
-    j <- Future.successful(result.getArray("results").get[JsonArray](0))
-    _ <- t.commit()
-  } yield j
+                              |  ORDER BY column_id""".stripMargin, Json.arr(tableId, columnId))
+  } yield result.get[JsonArray](0)
 
-  def getAll(table: Table): Future[JsonArray] = for {
-    t <- transaction.begin()
-    (t, result) <- t.query("""
-                              |SELECT column_id, user_column_name, column_type
-                              |  FROM system_columns
-                              |  WHERE table_id = ? ORDER BY column_id""".stripMargin, Json.arr(table.id))
-    j <- Future.successful {
-      result.getArray("results")
-    }
-    _ <- t.commit()
-  } yield j
+  def getAll(tableId: IdType): Future[JsonArray] = {
+    connection.singleQuery("SELECT column_id, user_column_name, column_type FROM system_columns WHERE table_id = ? ORDER BY column_id", Json.arr(tableId))
+  }
 
   def getToColumn(tableId: IdType, columnId: IdType): Future[JsonArray] = for {
-    t <- transaction.begin()
-    (t, result) <- t.query("""
+    result <- connection.singleQuery("""
                               |SELECT table_id, column_id
                               |  FROM system_columns
                               |  WHERE table_id != ? AND column_id != ? 
                               |  ORDER BY column_id""".stripMargin, Json.arr(tableId, columnId))
-    j <- Future.successful {
-      result.getArray("results").get[JsonArray](0)
-    }
-    _ <- t.commit()
-  } yield j
+  } yield result.get[JsonArray](0)
 
   def delete(tableId: IdType, columnId: IdType): Future[Unit] = for {
-    t <- transaction.begin()
-    (t, _) <- t.query(s"ALTER TABLE user_table_$tableId DROP COLUMN IF EXISTS column_$columnId", Json.arr())
-    (t, _) <- t.query("DELETE FROM system_columns WHERE column_id = ? AND table_id = ?", Json.arr(columnId, tableId))
+    t <- connection.begin()
+    (t, r) <- t.query(s"ALTER TABLE user_table_$tableId DROP COLUMN column_$columnId", Json.arr())
+    (t, re) <- t.query("DELETE FROM system_columns WHERE column_id = ? AND table_id = ?", Json.arr(columnId, tableId))

```

Comment:
If you don't need `r` and `re`, you can leave `_` there. Otherwise, I'd propose `r0`, `r1`, `r2`, etc. for a nicer alignment ;)